### ISSUE TITLE
Direct lighting still visible in reflections even when it's disabled

### ISSUE THREAD

**Issue by usakhelo** [appleseedhq/appleseed#2861]
Hi,
This issue is visible when non-physical lights are used.
There are spot lights in this scene but direct lighting is disabled. Nonetheless shadows and lighting is visible in reflection:
![image](https://user-images.githubusercontent.com/1850877/81123017-1dbe3c80-8f00-11ea-9bac-48d00d6eee10.png)

I think it comes from this line in `on_scatter` function in ptlightingengine.cpp
```
if (m_params.m_enable_dl || vertex.m_path_length > 1)
{
```

**Comment by oktomus** [comment]
Visible in reflection and direct lighting are two different things right ?

**Comment by Mango-3** [comment]
Yes, I think the definition of direct lighting is lighting provided from a source without reflection from other surfaces.
